models/quix/optimism/quix_v2_optimism_events.sql

```diff
@@ -0,0 +1,237 @@
+      and er.block_number = erc20.evt_block_number 
+      and erc20.value is not null 
+      and erc20.to not in (
+        lower('{{quix_fee_address_address}}') --qx platform fee address
+        ,er.seller
+        ,er.project_contract_address
+      )
+      {% if not is_incremental() %}
+      -- smallest block number for source tables above
+      and erc20.evt_block_number > '{{min_block_number}}'
+      {% endif %}
+      {% if is_incremental() %}
+      and erc20.evt_block_time >= date_trunc("day", now() - interval '1 week')
+      {% endif %}
+)
+select
+    'optimism' as blockchain
+    ,'quix' as project
+    ,'v2' as version
+    ,TRY_CAST(date_trunc('DAY', er.block_time) AS date) AS block_date
+    ,er.block_time
+    ,er.token_id 
+    ,n.name as collection
+    ,er.amount_raw / power(10, t1.decimals) * p1.price as amount_usd
+    ,'erc721' as token_standard
+    ,'Single Item Trade' as trade_type
+    ,cast(1 as bigint) as number_of_items
+    ,'Buy' as trade_category
+    ,'Trade' as evt_type
+    ,er.seller
+    ,case 
+    when er.buyer = agg.contract_address then erct2.to
+    else er.buyer 
+    end as buyer
+    ,er.amount_raw / power(10, t1.decimals) as amount_original
+    ,er.amount_raw
+    ,case 
+        when (erc20.contract_address = '0x0000000000000000000000000000000000000000' or erc20.contract_address is null)
+            then 'ETH'
+            else t1.symbol
+        end as currency_symbol
+    ,case 
+        when (erc20.contract_address = '0x0000000000000000000000000000000000000000' or erc20.contract_address is null) 
+            then '0xdeaddeaddeaddeaddeaddeaddeaddeaddead0000'
+            else erc20.contract_address
+        end as currency_contract
+    ,er.nft_contract_address
+    ,er.project_contract_address
+    ,agg.name as aggregator_name
+    ,agg.contract_address as aggregator_address
+    ,er.tx_hash
+    ,er.block_number
+    ,tx.from as tx_from
+    ,tx.to as tx_to
+    ,ROUND((2.5*(er.amount_raw)/100),7) as platform_fee_amount_raw
+    ,ROUND((2.5*((er.amount_raw / power(10,t1.decimals)))/100),7) AS platform_fee_amount
+    ,ROUND((2.5*((er.amount_raw / power(10,t1.decimals)* p1.price))/100),7) AS platform_fee_amount_usd
+    ,'2.5' as platform_fee_percentage
+    ,tr.value as royalty_fee_amount_raw
+    ,tr.value / power(10, t1.decimals) as royalty_fee_amount 
+    ,tr.value / power(10, t1.decimals) * p1.price as royalty_fee_amount_usd
+    ,(tr.value / er.amount_raw * 100) as royalty_fee_percentage
+    ,case when tr.value is not null then tr.to end as royalty_fee_receive_address
+    ,case when tr.value is not null
+        then case when (erc20.contract_address = '0x0000000000000000000000000000000000000000' or erc20.contract_address is null) 
+            then 'ETH' else t1.symbol end
+        end as royalty_fee_currency_symbol
+    ,'quix' || '-' || er.tx_hash || '-' || er.token_id || '-' || er.seller || '-' || coalesce(erct2.evt_index::string, '') as unique_trade_id   
```

Comment:
we're looking into removing the concept of ```unique_trade_id``` -- which would mean determining which fields make up the unique key combination and applying in the config block at the top (similar to all dex trades models). then also a test applied in the schema file (examples in dex again).

i don't want to block this PR though, @augustog do you think it's too soon to push for that here? this PR stops at protocol-level and doesn't go into nft trades, so may be a good time to start.

Reply:
happy to make it inside of the config if that's the way moving forward

Reply:
i see you pushed a few tries, let me see if we can get it to work as expected